### PR TITLE
Logan Williams' Updated Intro to GitHub

### PR DESCRIPTION
**Pull request by DisarmedRemo66**
Here is a new pull request with my Power of Thor assignment with the process of accepting revisions from both Dr. Jones and Logan Johnson.

### CODE REVIEW

File: lw1551/Power of Thor intro.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+// *************
+// This is my solution for the `Power of Thor coding game <https://www.codingame.com/training/easy/power-of-thor-episode-1/solution>`_
+//
+//
+// =====================
+// Thought Process
+// =====================
+// * Thor needs to be able to move based on the different locations of the power orb.
+// * Thor currently just walks off the screen and goes out of bounds due to being stuck in a while loop that does not check to see thor's location in relation to either the light or the border of the screen.
+// * There is going to need to be a way to increment Thor’s location in at least two variables depending on the cardinal direction he \moves in.
```
Comment (bjones1) [inline lw1551/Power of Thor intro.cs]: Why `\moves` instead of `moves`?

File: lw1551/Power of Thor intro.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+//
+// 2. Thor cannot run out of energy; the game allows for a **maximum of 100 energy**.
+//
+// ---------------------
+// Defined Variables
+// ---------------------
+//
+        string[] inputs = Console.ReadLine().Split(' ');
+        int lightX = int.Parse(inputs[0]); 
+        // The X position of the light of power
```
Comment (bjones1) [inline lw1551/Power of Thor intro.cs]: Put a comment before the line it describes, not after.

File: lw1551/Power of Thor intro.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        int lightY = int.Parse(inputs[1]); 
+        // The Y position of the light of power
+        int initialTX = int.Parse(inputs[2]); 
+        // Thor's starting X position
+        int initialTY = int.Parse(inputs[3]); 
+        // Thor's starting Y position
+ //
```
Comment (bjones1) [inline lw1551/Power of Thor intro.cs]: Indent this comment, so it matches the other comments.

File: lw1551/Power of Thor intro.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        // The first two variables, lightX and lightY, are used to set where the coordinates of the light orb's location for each new level of the game. I then defined initialTX and initialTY to be Thor's initial location. When first designing this program, I had two addtional variables that would update to thor's new location, but I realized when working through this that the initialTX and initialTY variables can be used for both functions.
```
Comment (bjones1) [inline lw1551/Power of Thor intro.cs]: Use a `monospaced` font for variables:

```
// The first two variables, ``lightX`` and ``lightY``, ...
```

Comment (bjones1) [reply lw1551/Power of Thor intro.cs]: Also, put this description before the variables.

File: lw1551/Power of Thor intro.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+//
+//
+//
+        string directionX = "";
+        // Blank X direction string to fill with direction thor will move in.
+        string directionY = "";
+        // Blank Y direction string to fill with direction thor will move in.
+        // These variables are used to read in the cardinal direction given by the player of the game.
```
Comment (bjones1) [inline lw1551/Power of Thor intro.cs]: Description before variables.

File: lw1551/Power of Thor intro.cs [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        // The X position of the light of power
+        int lightY = int.Parse(inputs[1]); 
+        // The Y position of the light of power
+        int initialTX = int.Parse(inputs[2]); 
+        // Thor's starting X position
+        int initialTY = int.Parse(inputs[3]); 
+        // Thor's starting Y position
+ //
+        // The first two variables, lightX and lightY, are used to set where the coordinates of the light orb's location for each new level of the game. I then defined initialTX and initialTY to be Thor's initial location. When first designing this program, I had two addtional variables that would update to thor's new location, but I realized when working through this that the initialTX and initialTY variables can be used for both functions.
+//
```
Comment (bjones1) [inline lw1551/Power of Thor intro.cs:38]: Indent.

File: lw1551/Power of Thor intro.cs [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        string directionX = "";
+        // Blank X direction string to fill with direction thor will move in.
+        string directionY = "";
+        // Blank Y direction string to fill with direction thor will move in.
+        // These variables are used to read in the cardinal direction given by the player of the game.
+
+// ****************
+// Solution
+// ****************
+// This code operates by making multiple comparisons each time Thor moves. The loop begins by establishing that thor has a X and a Y direction, and creating a string variable that will store the Direction in it. For example, if Thor needs to move to the bottom right, then the if statements will check for whether Thor's X location is larger than the light's X value. The same will be done for the Y values of the game. Therefore after several runs through the code you will be moving in all cardinal directions. So the basic idea is that the strings will tell which cardinal direction we want to go in, while the code itself will run comparisons between thor's orginal/updated location and the light's location to decide if the move is valid for the game to be won.
```
Comment (bjones1) [inline lw1551/Power of Thor intro.cs]: Show a map / image / diagram that explains how compass directions relate to x and y coordinates.

File: lw1551/Power of Thor intro.cs [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+        // Blank X direction string to fill with direction thor will move in.
+        string directionY = "";
+        // Blank Y direction string to fill with direction thor will move in.
+        // These variables are used to read in the cardinal direction given by the player of the game.
+
+// ****************
+// Solution
+// ****************
+// This code operates by making multiple comparisons each time Thor moves. The loop begins by establishing that thor has a X and a Y direction, and creating a string variable that will store the Direction in it. For example, if Thor needs to move to the bottom right, then the if statements will check for whether Thor's X location is larger than the light's X value. The same will be done for the Y values of the game. Therefore after several runs through the code you will be moving in all cardinal directions. So the basic idea is that the strings will tell which cardinal direction we want to go in, while the code itself will run comparisons between thor's orginal/updated location and the light's location to decide if the move is valid for the game to be won. 
+            while (true)
```
Comment (bjones1) [inline lw1551/Power of Thor intro.cs]: Wrong indent

File: lw1551/Power of Thor intro.cs [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+        {
+            int remainingTurns = int.Parse(Console.ReadLine()); 
+            // The remaining amount of turns Thor can move. Do not remove this line.
+
+            string directionX = "";
+            // Blank X direction string to fill with direction thor will move in.
+            string directionY = "";
+            // Blank Y direction string to fill with direction thor will move in.
+
+            if (initialTX > lightX)
```
Comment (bjones1) [inline lw1551/Power of Thor intro.cs]: Explain --- why if / if else, then another if / if else pair?

File: lw1551/Power of Thor intro.cs [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+            if(initialTY > lightY)
+            {
+                directionY = "N";
+                initialTY--;
+            }
+            else if(initialTY < lightY)
+            {
+                directionY = "S";
+                initialTY++;
+            }
```
Comment (bjones1) [inline lw1551/Power of Thor intro.cs]: Does this work? I don't see any place where your program output the direction to move in.

**Review by bjones1 — CHANGES_REQUESTED** [review]

**Comment by bjones1** [comment]
I haven't seen any progress on this. Message me after you push update and are ready for another review.

**Comment by DisarmedRemo66** [comment]
I have pushed a new update with the changes you recommended.

File: lw1551/Power of Thor intro.cs [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+        // The X position of the light of power
+        int lightX = int.Parse(inputs[0]); 
+        // The Y position of the light of power
+        int lightY = int.Parse(inputs[1]); 
+        // Thor's starting X position
+        int initialTX = int.Parse(inputs[2]); 
+        // Thor's starting Y position
+        int initialTY = int.Parse(inputs[3]); 
+        // The first two variables, ``lightX`` and ``lightY``, are used to set where the coordinates of the light orb's location for each new level of the game. I then defined ``initialTX`` and ``initialTY`` to be Thor's initial location. When first designing this program, I had two addtional variables that would update to thor's new location, but I realized when working through this that the ``initialTX`` and ``initialTY`` variables can be used for both functions.
+        string directionX = "";
```
Comment (bjones1) [inline lw1551/Power of Thor intro.cs]: These variables make no sense here, since you re-define them a few lines later. Remove.

**Review by bjones1 — CHANGES_REQUESTED** [review]

File: lw1551/Power of Thor intro.cs (hunk 11 above)
Comment (DisarmedRemo66) [reply lw1551/Power of Thor intro.cs]: I have removed these lines and updated comments for clarity purposes.

**Pull request merged**